docs/rule/no-obscure-array-access.md

```diff
@@ -0,0 +1,40 @@
+# no-obscure-array-access
+
+Using obscure expressions `{{list.[1].name}}` is discouraged and is likely to be deprecated soon.
```

Comment:
We should mention that this break Prettier too iirc (https://github.com/prettier/prettier/issues/4908#issuecomment-682125029)

There's also a codemod to help https://github.com/mahirshah/ember-array-index-template-codemod

Reply:
Wow nice thanks for this.

Reply:
From [the discord](https://discord.com/channels/480462759797063690/486548111221719040/542754258689458197) it seems like the recommended way to access the array index is:

```
{{get (get users 0) 'name'}}
```
Nesting the array index getter inside the non array path getter, that is how I implemented the fixer codemod for this rule. If we don't need to nest it makes the fixer much simpler and I can do basically what that codemod is doing.

Reply:
For prettier, it doesn't break when I try it locally, it just converts `[0]` to `0`.

Reply:
Ah nice maybe that was fixed it used to break